CMake/mlpack.cmake

```diff
@@ -0,0 +1,682 @@
+#     stb/ directory; if "YES", then includes must be done as, e.g.,
+#     stb/stb_image.h.
+#
+#
+# find_openmp
+#-----------------------
+#
+# This macro finds if OpenMP library is installed and supported by the
+# compiler. if the library found then it sets the following parameters:
+#  
+# OpenMP_FOUND - set to true if the library is found
+# 
+#
+# find_mlpack_internal
+#-----------------------
+#
+# This macro finds the mlpack library and sets the necessary paths to each one
+# of the parameters. If the library is not found this macro will set
+# MLPACK_FOUND to false.
+#
+# This module sets the following variables:
+#  MLPACK_FOUND - set to true if the library is found
+#  MLPACK_VERSION_MAJOR - major version number
+#  MLPACK_VERSION_MINOR - minor version number
+#  MLPACK_VERSION_PATCH - patch version number
+#  MLPACK_VERSION_STRING - version number as a string (ex: "1.0.4")
+#  MLPACK_INCLUDE_DIR - list of mlpack include directories
+#
+# fetch_mlpack
+#-----------------------
+#
+# This function downloads the mlpack library and its dependencies by calling
+# the get_deps function. This will allow to download each one of the libraries
+# and set the parameter for each one of them corectly.
+#
+# fetch_mlpack accepts one parameter which is COMPILE_OPENBLAS. This will allow
+# the user to compile OpenBLAS after downloading it since it is the only
+# library that requires compilation in our toolchain, all other libraries are
+# header only. 
+#
+# fetch_mlpack will download mlpack library as well, so it can be used directly
+# in cmake if the user would like to download the latest version or not use the
+# one already exist on the system.
+#
+#
+# find_mlpack 
+#----------------------
+#
+# This macro finds mlpack libraries and it dependencies. This is the function
+# that should be called by the user, and it will call all the other dependency
+# functions. This function will set MLPACK_FOUD to false if the library is not
+# found locally through find_mlpack_internal
+#
+# This macro will set the following variables:
+# 
+# MLPACK_INCLUDE_DIRS - list of all include directories
+# MLPACK_LIBRARIES - list of all libraries to link against.
+#
+##===================================================
+##  MLPACK DEPEDENCIES SETTINGS. 
+##===================================================
+
+# Set minimum library versions required by mlpack.
+#
+# For Armadillo, try to keep the minimum required version less than or equal to
+# what's available on the current Ubuntu LTS or most recent stable RHEL release.
+# See https://github.com/mlpack/mlpack/issues/3033 for some more discussion.
+set(ARMADILLO_VERSION "10.8.2")
+set(ENSMALLEN_VERSION "2.10.0")
+set(CEREAL_VERSION "1.1.2")
+set(OPENBLAS_VERSION "0.3.29")
+
+# Set required standard to C++17.
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+set(MLPACK_DISABLE_OPENMP OFF)
+
+if (NOT USE_OPENMP)
+  set(MLPACK_DISABLE_OPENMP ON)
+endif()
```

Comment:
I think we can just use `MLPACK_DISABLE_OPENMP` in this file.  This bit can go in the main `CMakeLists.txt` to translate the option name `USE_OPENMP` to the name `MLPACK_DISABLE_OPENMP`; see the comment I left in that file.